drivers/sensor/aht20/aht20.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2022 Rafael Lee <rafaellee.img@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT asair_aht20
+
+#include <kernel.h>
+#include <device.h>
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+#include <logging/log.h>
+#include <sys/crc.h>
+#include <sys/byteorder.h>
+
+#include "aht20.h"
+
+LOG_MODULE_REGISTER(AHT20, CONFIG_SENSOR_LOG_LEVEL);
+
+static int aht20_channel_get(const struct device *dev, enum sensor_channel chan,
+			     struct sensor_value *val)
+{
+	struct aht20_data *drv_data = dev->data;
+	float temperature;
+
+	switch (chan) {
+	case SENSOR_CHAN_HUMIDITY:
+		val->val1 = 0;
+		val->val2 = (drv_data->humidity) * 1000000LL / (1 << AHT20_FULL_RANGE_BITS);
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		temperature = (float)drv_data->temperature / (1 << AHT20_FULL_RANGE_BITS) *
+			      AHT20_TEMPERATURE_RANGE -
+			      AHT20_TEMPERATURE_OFFSET;
+		val->val1 = (int32_t)temperature;
+		val->val2 = (temperature - val->val1) * 1000000LL;
+		break;
+	default:
+		break;
+	}
+	return 0;
+}
+
+static int aht20_init(const struct device *dev)
+{
+	struct aht20_data *drv_data = dev->data;
+
+	k_sleep(K_MSEC(40)); // wait for 40ms
+
+	int ret = 0;
+	aht20_status sensor_status = { 0 };
+
+	static uint8_t const inquire_status_seq[] = { AHT20_CMD_GET_STATUS };
+
+	ret = i2c_write_dt(&drv_data->bus, inquire_status_seq, sizeof(inquire_status_seq));
```

Comment:
i2c_write_read_dt doesn't work for this sequence of doing a GET_STATUS and obtaining the status value?